File: db/ipfsdb.js
import pycurl
import certifi
from io import BytesIO

c = pycurl.Curl()
#因为我本地的vpn转发端口在1080，可更换
c.setopt(pycurl.PROXY,'127.0.0.1:1080')
c.setopt(pycurl.CAINFO, certifi.where())

def get(DataHash):
    url = f'https://ipfs.infura.io:5001/api/v0/get?arg={DataHash}&archive=true'
    c.setopt(pycurl.URL, url)
    try:
        re =  c.perform()
    except:
        return null
    return  re



if __name__ == '__main__':
    print(get('QmZtmD2qt6fJot32nabSP3CUjicnypEBz7bHVDhPQt9aAy')) 

File: db/ipfsdb.py
import requests
import json
proxies = {
    "http": "http://127.0.0.1:1080",
    "https": "127.0.0.1:1080",
}
'''
this function search hash in IPFS network and return the content required by hash
input:  Hash:String
output: Content:Json
'''
def get(Hash):

    try:
        re=requests.get("https://ipfs.infura.io:5001/api/v0/cat?arg={Hash}",proxies=proxies)
    except:
        return 0

    return re.content.json()

'''
This function upload file required by the file path and return the file hash
input: Path:String
output: Hash:String
'''
def add(Path):
    url = f'https://ipfs.infura.io:5001/api/v0/add?pin=false'
    payload = {'file':Path}
    headers =  {'Content-Type':'multipart/form-data'}
    p=requests.post(url=url,data=payload,headers=headers)
    print(p.url)
    print(p.content)

if __name__ == '__main__':
    add(f'D:\\1.json')
